appcues/src/main/java/com/appcues/statemachine/State.kt

```diff
@@ -3,42 +3,32 @@ package com.appcues.statemachine
 import com.appcues.data.model.Experience
 import com.appcues.util.ResultOf
 
-internal sealed class State {
+internal sealed class State(open val experience: Experience) {
 
-    object Idling : State()
-    data class BeginningExperience(val experience: Experience) : State()
+    data class Idling(override val experience: Experience) : State(experience)
```

Comment:
whenever we instanciate a new StateMachine, it will be tied to a particular experience. when the Experience is no longer visible, moves to idle or throws an error, we still know which experience it is.

this helps when we need to get the experience.renderContext when filtering which states should be sent to a particular subscriber from ExperienceRenderer

Reply:
@andretortolano the lightbulb moment for me was realizing that we need a state machine for each current render context (always a modal one running, and perhaps more for embed views), not each experience.

I also started by doing it with state machine associated with a particular experience, but realized it was a much nicer (conceptually and in the code). My initial thoughts are [here](https://appcues.slack.com/archives/C031RLGS4F8/p1686832146095029).

Reply:
I'm not sure I follow, I understand there is an issue related to frameId not being present when an embedded PresenterTrait tries to present(), so to avoid that I was imagining that ExperienceRenderer would manage the experiences that are being displayed or not somehow. then the embeds that dont match a registered view would be stored in a cache list without being started, then when a Appcues.registerFrame(frameId) gets called we would check if that frameId represents a frameId we have a cached experience for.

to james argument I initially thought that there would be no difference between a modal renderContext and an embed, aside from the fact that Modal would be unique where Embed(frameId) can have multiple (1 for each frameId string)